### DISCUSSION TITLE
Submit Getting more return values ​​than I intended.

### DISCUSSION THREAD

**Discussion by kimmoonil** [react-hook-form/react-hook-form#9373] [Q&A] [answered]
When I add input dynamically, this problem is happend.

e.g Dynamically Adding

```
const methods = useFormContext();
const [ state , setState ] = useState([])
// put the value to state array 

...some codes ... 

return (
   {state.map((item) => <input {...methods.register(randomId())} />))}
)
```
Input.tsx

```
const methods = useForm<string[]>()
const { handleSubmit } = methods
const onSubmit(data : any) => {
      console.log(data)
}

... some codes ... 

return (
<FormProvider {...methods}>
     <form onSubmit={handleSubmit(onSubmit)}>
      ...
)
  
```
InputParents.tsx

I thought console.log(data) would be equal to state length.

<img width="363" alt="image" src="https://user-images.githubusercontent.com/26863276/202243489-30c49194-e12a-429b-a547-4eea3e5b9e8c.png">

i have only 6 ( 3 * 2 )  input , but data has 12. 
( my first form input is 1 , and second is 2 . The first value is repeated three times.  )

**Comment by bluebill1049** [comment, marked as answer]
each register invoke will update the internal store for an input value, you either have to make sure when component gets unmounted call `unregister` which you can simplify create a component and do that at useEffect return callbac k
